portfolio/src/main/webapp/script.js

```diff
@@ -17,22 +17,38 @@ function addNav() {
   replaceIdWithQuery("nav", "navbar.html");
 }
 
-function replaceIdWithQuery(id, url) {
+function addContentToId(id, url, updateUrl) {
+  if (!this.requested) {
+    this.requested = new Set();
+  }
+
+  if (this.requested.has(updateUrl)) {
+    document.getElementById(updateUrl).style.display = "block";
+    return;
+  }
+
   var xhttp = new XMLHttpRequest();
 
-  var success = false;
   xhttp.onreadystatechange = function() {
-    if (this.readyState === 4 && this.status === 200) {
-      document.getElementById(id).innerHTML = this.responseText;
-      success = true;
+    if (this.readyState === 4) {
```

Comment:
What is the significance of 4?

Reply:
4 means the request was completed. I changed it to use a preexisting class variable to make it clearer.